### DISCUSSION TITLE
Interactive app- WebBrowser view for outputes of type HTML

### DISCUSSION THREAD

**Discussion by glemis** [NeilMacMullen/Textrude#225] [Ideas]
I personally made a panel for HTML outputs to show it in real time rendering. 
I use this to author web content and HTML that is pushed into PDFs and being able to view and style in real time is invaluable. 

Wondering your thoughts on how it could best be put into the view as my approach is lazy and just adding a 4th column which wouldn't work for people who don't have a big screen like me. 

I think potentally something along the lines of a menu option to open a new window with the HTML in it maybe with config options in app for widow dimensions (as in my case we would love to make it set to the size the PDF's are so it is very representitive of final result, 
Also this way users with 2 monitors can have outputs on second while autoring on first or tab between on single monitor.

**Comment by NeilMacMullen** [comment]
I think there are actually two separate ideas here... 

1.  Allow more flexibility in the UI.  (I have 2 monitors myself so feel your frustration!) Ideally any output pane or collection of panes should be draggable outside the main window.  I'm _very_ open to PRs that improve the UI :-)

2. Allow display of rendered HTML (and markdown would be nice!)  rather than syntax coloring of HTML tags.   The  output pane uses the Monaco editor which is hosted in a WebView2 control.  In order of preference (best first), potential mechanisms for rendering HTML would be....
  - use built-in rendering from Monaco in order to piggy-back on development effort there.  _Unfortunately I'm not sure that Monaco has a "highlighter" that performs this kind of rendering_
  - Figure out a way to switch the WebView contents between Monaco JS and raw HTML when the user wants that
  - Modify OutputPane to overlay a separate HTML renderer control on top of the WebView when the user wants that.
  - Create a new "render as" (HTML/Markdown/PDF) output concept off in a separate area of the application
So ideally, the "rendered HTML" option should just appear as an extra option in the long list of supported highlighters and the user should be able to toggle between raw HTML and rendered.  

Having said that, I'm open to the ideas that "rendered" output (HTML/Markdown/PDF etc) should be treated differently - the current output tab design really is a direct binding to output files.  Once you might want to (possibly simultaneously) see multiple representations of the same output file (e.g. raw HTML and rendered) it starts to make a bit more sense to introduce a "render" area.   And it's probably easier to implement since you don't have to follow the plumbing associated with the WebView stuff.
